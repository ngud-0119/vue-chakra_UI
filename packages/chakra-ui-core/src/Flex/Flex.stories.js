import { storiesOf } from '@storybook/vue'
import { Flex, Box, Text as KText } from '..'

storiesOf('UI | Flex', module)
  .add('Flex', () => ({
    components: { Flex, Box, KText },
    template: `
      <div>
      <Flex align="center">
        <Flex bg="green.50" align="flex-end">
          <KText>Box 1</KText>
        </Flex>
        <Flex bg="blue.50" size="150px" align="center" justify="center">
          <KText textAlign="center" bg="orange.50">
            Box 2
          </KText>
        </Flex>
        <Box>
          <KText bg="tomato" color="white">
            Box 3
          </KText>
        </Box>
      </Flex>
      </div>
    `
  }))
