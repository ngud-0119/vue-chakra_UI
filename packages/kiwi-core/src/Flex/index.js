import Box from '../Box'
import { baseProps } from '../config/props'
import { forwardProps } from '../utils'

/**
 * Flex is Box with display: flex and comes with helpful style shorthand. It renders a div element.
 */
const Flex = {
  name: 'Flex',
  props: {
    align: [String, Array],
    justify: [String, Array],
    wrap: [String, Array],
    direction: [String, Array],
    size: [String, Array],
    ...baseProps
  },
  render (h) {
    return h(Box, {
      props: {
        display: 'flex',
        flexDirection: this.direction,
        alignItems: this.align,
        justifyContent: this.justify,
        flexWrap: this.wrap,
        h: this.size,
        w: this.size,
        ...forwardProps(this.$props)
      }
    }, this.$slots.default)
  }
}

export default Flex
