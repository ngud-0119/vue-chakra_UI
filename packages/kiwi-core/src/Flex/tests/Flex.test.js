import Flex from '../'
import { render } from '@/tests/test-utils'

const renderComponent = (props) => {
  const inlineAttrs = (props && props.inlineAttrs) || ''
  const base = {
    components: { Flex },
    template: `<Flex ${inlineAttrs}>Flex Me</Flex>`,
    ...props
  }
  return render(base)
}

it('should render correctly', () => {
  const { asFragment } = renderComponent()

  expect(asFragment()).toMatchSnapshot()
})

it('should change orientation', () => {
  const inlineAttrs = 'align="center"'
  const { asFragment } = renderComponent({ inlineAttrs })

  expect(asFragment()).toMatchSnapshot()
})
