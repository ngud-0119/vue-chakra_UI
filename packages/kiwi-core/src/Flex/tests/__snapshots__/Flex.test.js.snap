// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change orientation 1`] = `
<DocumentFragment>
  <div
    class="css-70qvj9"
  >
    Flex Me
  </div>
</DocumentFragment>
`;

exports[`should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-k008qs"
  >
    Flex Me
  </div>
</DocumentFragment>
`;
